packages/block-editor/src/utils/dom.js

```diff
@@ -156,11 +154,24 @@ export function getVisibleElementBounds( element ) {
 		for ( const child of currentElement.children ) {
 			if ( isElementVisible( child ) ) {
 				const childBounds = child.getBoundingClientRect();
-				bounds = rectUnion( bounds, childBounds, viewportRect );
+				bounds = rectUnion( bounds, childBounds );
 				stack.push( child );
 			}
 		}
 	}
 
+	/*
+	 * Take into account the outer horizontal limits of the container in which
+	 * an element is supposed to be "visible". For example, if an element is
+	 * positioned -10px to the left of the window x value (0), this function
+	 * discounts the negative overhang because it's not visible and therefore
+	 * not to be counted in the visibility calculations. Top and bottom values
+	 * are not accounted for to accommodate vertical scroll.
+	 */
+	bounds = rectIntersect(
+		bounds,
+		new window.DOMRect( 0, -Infinity, viewport.innerWidth, Infinity )
```

Comment:
Did you test this with `Infinity?` What's the reason behind that value? It seems to have reverted the efforts in  https://github.com/WordPress/gutenberg/pull/62711 😄 

<img width="536" alt="Screenshot 2024-08-28 at 11 24 07 AM" src="https://github.com/user-attachments/assets/6f5cdc47-a5c6-409f-bf26-47b1d266e2c4">

I thought we were interested in visibility, so defined bounds on the container would work but I could be missing something

`new window.DOMRect( 0, 0, viewport.innerWidth, viewport.innerHeight )`

Reply:
Didn't the original code only clip along the X axis?

Reply:
Ah yes, true 🤔 

I'll withdraw that, and speculate that consumers of `getVisibleElementBounds` don't know how to deal with `NaN` values being returned here (?)

Reply:
Hmm ok. Forget `rectIntersect` then, we can just adjust `left` and `right` manually. I still think it's nice to not do this in `rectUnion` though. Pushed.